apps/meteor/ee/app/livechat-enterprise/server/hooks/resumeOnHold.ts

```diff
@@ -1,46 +1,68 @@
+	const {
+		v: { _id: visitorId },
+		_id: rid,
+	} = room;
+	const visitor = await LivechatVisitors.findOneById<Pick<ILivechatVisitor, 'name' | 'username'>>(visitorId, {
+		projection: { name: 1, username: 1 },
+	});
+	if (!visitor) {
+		callbackLogger.error(`[afterSaveMessage] Visitor Not found for room ${rid} while trying to resume on hold`);
+		throw new Error('Visitor not found while trying to resume on hold');
+	}
 
-		// if the message has a type means it is a special message (like the closing comment), so skips
-		if (message.t) {
-			return message;
-		}
+	const guest = visitor.name || visitor.username;
 
-		if (!isOmnichannelRoom(roomParams)) {
-			return message;
-		}
+	const resumeChatComment = i18n.t('Omnichannel_on_hold_chat_automatically', { guest });
 
-		const { _id: rid, t: roomType, v: roomVisitor } = roomParams;
+	const resumedBy = await Users.findOneById('rocket.cat');
+	if (!resumedBy) {
+		callbackLogger.error(`[afterSaveMessage] User Not found for room ${rid} while trying to resume on hold`);
+		throw new Error(`User not found while trying to resume on hold`);
+	}
 
-		// message valid only if it is a livechat room
-		if (!(typeof roomType !== 'undefined' && roomType === 'l' && roomVisitor && roomVisitor.token)) {
-			return message;
-		}
+	return { comment: resumeChatComment, resumedBy };
+};
+
+const handleAfterSaveMessage = async (message: IMessage, room: IRoom) => {
+	if (isEditedMessage(message) || message.t || !isOmnichannelRoom(room)) {
+		return message;
+	}
+
+	const { _id: rid, v: roomVisitor } = room;
 
-		// Need to read the room every time, the room object is not updated
-		const room = await LivechatRooms.findOneById(rid, { projection: { t: 1, v: 1, onHold: 1 } });
-		if (!room) {
+	if (!roomVisitor?._id) {
+		return message;
+	}
+
+	// Need to read the room every time, the room object is not updated
+	const updatedRoom = await LivechatRooms.findOneById(rid);
+	if (!updatedRoom) {
+		return message;
+	}
+
+	if (message.token && room.onHold) {
+		callbackLogger.debug(`[afterSaveMessage] Room ${rid} is on hold, resuming it now since visitor sent a message`);
+
+		try {
+			const { comment: resumeChatComment, resumedBy } = await resumeOnHoldCommentAndUser(updatedRoom);
+			await OmnichannelEEService.resumeRoomOnHold(updatedRoom, resumeChatComment, resumedBy);
+		} catch (error) {
+			callbackLogger.error(`[afterSaveMessage] Error while resuming room ${rid} on hold: Error: `, error);
 			return message;
 		}
+	}
 
-		// if a visitor sends a message in room which is On Hold
-		if (message.token && room.onHold) {
-			await Meteor.callAsync('livechat:resumeOnHold', rid, { clientAction: false });
-		}
+	return message;
+};
 
-		return message;
-	},
-	callbacks.priority.HIGH,
-	'livechat-resume-on-hold',
-);
+callbacks.add('afterSaveMessage', handleAfterSaveMessage, callbacks.priority.HIGH, 'livechat-resume-on-hold');
```

Comment:
Was this role re-created by some conflict resolution?

Reply:
Sorry I didn't understood what you meant here. Could you please elaborate?